### PR TITLE
PVLV update

### PR DESCRIPTION
**Pull request by rcoreilly**
This is the big PVLV update.  The PVLV part itself is good to go for now (still more to test and tune) but it is not yet integrated and tested with BOA, and it needs just a few more minor things...  anyway, review can start now, and PVLV.md has all the docs updated.

### CODE REVIEW

File: examples/boa/boa.go [hunk 1]
```diff
@@ -373,26 +370,26 @@ func (ss *Sim) ConfigNet(net *axon.Network) {
 	////////////////////////////////////////////////
 	// position
 
-	vPgpi.SetRelPos(relpos.Rel{Rel: relpos.RightOf, Other: rew.Name(), YAlign: relpos.Front, Space: space})
-	ach.SetRelPos(relpos.Rel{Rel: relpos.Behind, Other: snc.Name(), XAlign: relpos.Left, Space: space})
+	vPgpi.PlaceRightOf(rew, space)
```
Comment (etuleu) [inline examples/boa/boa.go:373]: This looks much nicer!

File: axon/drivepvlv.go [hunk 2]
```diff
@@ -0,0 +1,534 @@
...
+import (
+	"github.com/goki/gosl/slbool"
+	"github.com/goki/mat32"
+)
+
+//gosl: start drivepvlv
+
+// DriveVals represents different internal drives, such as hunger, thirst, etc.
+// labels can be provided by specific environments.
+type DriveVals struct {
```
Comment (etuleu) [inline axon/drivepvlv.go:16]: why not use an array?

Comment (rcoreilly) [reply axon/drivepvlv.go:16]: no arrays in GPU land..

File: axon/deep_net.go [hunk 3]
```diff
@@ -298,9 +298,9 @@ func (nt *Network) AddPTPredLayer(ptMaint, ct, thal *Layer, ptToPredPrjn, ctToPr
 	ptPred.PlaceBehind(ptMaint, space)
 	nt.ConnectCtxtToCT(ptMaint, ptPred, ptToPredPrjn).SetClass("PTtoPred")
 	nt.ConnectLayers(ct, ptPred, ctToPredPrjn, emer.PrjnType(ForwardPrjn)).SetClass("CTtoPred")
-	toThal, fmThal := nt.BidirConnectLayers(ptMaint, thal, toThalPrjn)
-	toThal.SetClass("PredToThal")
-	fmThal.SetClass("ThalToPred")
+	// toThal, fmThal := nt.BidirConnectLayers(ptPred, thal, toThalPrjn)
```
Comment (etuleu) [inline axon/deep_net.go:301]: do we want this to stay commented out?

File: examples/boa/boa.go [hunk 4]
```diff
@@ -337,16 +340,23 @@ func (ss *Sim) ConfigNet(net *axon.Network) {
 
 	net.ConnectLayers(dist, alm, full, emer.Forward)
 	net.ConnectLayers(effort, alm, full, emer.Forward)
-	net.ConnectLayers(ofcPT, alm, full, emer.Forward)
-	net.ConnectLayers(accPT, alm, full, emer.Forward)
+	// net.ConnectLayers(ofcPT, alm, full, emer.Forward)
```
Comment (etuleu) [inline examples/boa/boa.go:343]: same here?

File: examples/boa/boa.go [hunk 5]
```diff
@@ -576,9 +591,9 @@ func (ss *Sim) ConfigLoops() {
 	// GUI
 
 	if ss.Args.Bool("nogui") {
-		man.GetLoop(etime.Test, etime.Trial).Main.Add("NetDataRecord", func() {
-			ss.GUI.NetDataRecord(ss.ViewUpdt.Text)
-		})
+		// man.GetLoop(etime.Test, etime.Trial).Main.Add("NetDataRecord", func() {
```
Comment (etuleu) [inline examples/boa/boa.go:594]: also here?

**Review by etuleu — APPROVED** [review]
I left just a few comments next to commented out code just so you can take a look again to make sure it's not commented out by mistake. And maybe we can leave a note saying why it is commented out.

I ran it locally also and it looks good! 
Awesome work!

**Comment by etuleu** [comment]
Just an idea: maybe we should update the readme file (pvlv.md) with a section about the PTNotMaint layer? I might have missed it so will check it again, but just wanted to write this before I forgot.

File: axon/deep_net.go (hunk 3 above)
Comment (rcoreilly) [reply axon/deep_net.go:301]: good point -- just discovered this was being done late last night -- now removed.

Comment (rcoreilly) [reply axon/deep_net.go:301]: and added comment why

**Comment by rcoreilly** [comment]
thanks for the comments -- I made another pass through the ConfigNet and removed or commented on stuff that doesn't make sense, and added todo comments for all the stuff that still needs to be tested (later)..

I added the PTNotMaintLayer to DEEP.md (technically part of that framework) and to the boa README.md

File: PVLV.md [hunk 6]
```diff
@@ -1,33 +1,188 @@
...
+* Tobler, P. N., Dickinson, A., & Schultz, W. (2003). Coding of predicted reward omission by dopamine neurons in a conditioned inhibition paradigm. Journal of Neuroscience, 23, 10402–10410. http://www.ncbi.nlm.nih.gov/pubmed/14614099
+
+* Waelti, P., Dickinson, A., & Schultz, W. (2001). Dopamine responses comply with basic assumptions of formal learning theory. Nature, 412, 43–48. http://www.ncbi.nlm.nih.gov/pubmed/11452299
+
+
+# TODO:
+
+* multiplicative effort discounting will not produce a net negative on goal failure -- need an extra case for that.
+
+* PT not clearing -- need to clear super too!
```
Comment (etuleu) [inline PVLV.md:187]: Is this still an issue?

**Review by etuleu — APPROVED** [review]

**Pull request merged**